core/lib/contracts_loader/src/lib.rs

```diff
@@ -0,0 +1,135 @@
+pub async fn load_settlement_layer_contracts(
+    sl_client: &dyn EthInterface,
+    bridgehub_address: Address,
+    l2_chain_id: L2ChainId,
+    multicall3: Option<Address>,
+) -> anyhow::Result<Option<ChainSpecificContracts>> {
+    let gateway_diamond_proxy: Address =
+        CallFunctionArgs::new("getZKChain", Token::Uint(l2_chain_id.as_u64().into()))
+            .for_contract(bridgehub_address, &bridgehub_contract())
+            .call(sl_client)
+            .await?;
+
+    if gateway_diamond_proxy.is_zero() {
+        return Ok(None);
+    }
+
+    let ctm_address =
+        CallFunctionArgs::new("chainTypeManager", Token::Uint(l2_chain_id.as_u64().into()))
+            .for_contract(bridgehub_address, &bridgehub_contract())
+            .call(sl_client)
+            .await?;
+
+    let server_notifier_addr = CallFunctionArgs::new("serverNotifierAddress", ())
+        .for_contract(ctm_address, &state_transition_manager_contract())
+        .call(sl_client)
+        .await
+        .map(|a: Address| if a.is_zero() { None } else { Some(a) })?;
+
+    let validator_timelock_addr = CallFunctionArgs::new("validatorTimelock", ())
+        .for_contract(ctm_address, &state_transition_manager_contract())
+        .call(sl_client)
+        .await
+        .map(|a: Address| if a.is_zero() { None } else { Some(a) })?;
+
+    let chain_admin =
+        CallFunctionArgs::new("getChainAdmin", Token::Uint(l2_chain_id.as_u64().into()))
+            .for_contract(ctm_address, &state_transition_manager_contract())
+            .call(sl_client)
+            .await
+            .map(|a: Address| if a.is_zero() { None } else { Some(a) })?;
+
+    Ok(Some(ChainSpecificContracts {
+        ecosystem_contracts: EcosystemCommonContracts {
+            bridgehub_proxy_addr: Some(bridgehub_address),
+            state_transition_proxy_addr: Some(ctm_address),
+            server_notifier_addr,
+            validator_timelock_addr,
+            multicall3,
+        },
+        chain_contracts_config: ChainContracts {
+            diamond_proxy_addr: gateway_diamond_proxy,
+            chain_admin,
+        },
+    }))
+}
+
+// Load Contracts specific for l1, from bridgehub contracts
+pub async fn load_l1_specific_contracts(
+    eth_client: &dyn EthInterface,
+    bridgehub_address: Address,
+    l2_chain_id: L2ChainId,
+) -> anyhow::Result<L1SpecificContracts> {
+    let base_token = CallFunctionArgs::new("baseToken", Token::Uint(l2_chain_id.as_u64().into()))
+        .for_contract(bridgehub_address, &bridgehub_contract())
+        .call(eth_client)
+        .await
+        .map(|a: Address| if a.is_zero() { None } else { Some(a) })?;
+
+    let shared_bridge = CallFunctionArgs::new("sharedBridge", ())
+        .for_contract(bridgehub_address, &bridgehub_contract())
+        .call(eth_client)
+        .await
+        .map(|a: Address| if a.is_zero() { None } else { Some(a) })?;
+
+    let erc20_bridge = CallFunctionArgs::new("legacyBridge", ())
```

Comment:
there is no such function in bridgehub

Reply:
removed this function, will add somewhere in a future